core/patina_internal_collections/src/node.rs

```diff
@@ -204,6 +204,31 @@ where
         // When current capacity is 0, we just need to copy the data and build the available list
         if self.capacity() == 0 {
             self.data = buffer;
+
+            // Initialize all Cell fields to prevent reading uninitialized memory.
+            // Cell::set() internally uses mem::replace() which reads the old value before writing.
+            // We use ptr::write to initialize the fields without creating references to uninitialized data.
+            for i in 0..self.data.len() {
```

Comment:
@garybeihl 

I need to look at this more in depth, but if this is the solution, it should probably be completed in `Self::build_linked_list`. This function is already looping through each uninitialized node to set up the linked list of available nodes for allocations. A simple solution would just be to initialize it there while setting up the linked list. It also stops the need to writing null_ptr to the left and right nodes because we are setting them (except for the first and last node in the linked list)

Reply:
I had thought the same thing, but `build_linked_list` gets called when the list is expected to be initialized, it looks like.

Reply:
I think this code creates a mut ref to the slice without initializing the nodes within is technically UB already. So I don't think you can defer it to `build_linked_list` unless you defer creation of the slice to there. 
```
let buffer = unsafe {
            slice::from_raw_parts_mut::<'a, Node<D>>(
                slice as *mut [u8] as *mut Node<D>,
                slice.len() / mem::size_of::<Node<D>>(),
            )
        };
```

Reply:
@garybeihl can you post the actual full miri test failure?